yb-voyager/cmd/exportData.go

```diff
@@ -735,50 +747,105 @@ func reportUnsupportedTables(finalTableList []sqlname.NameTuple) {
-		tuple := sqlname.NameTuple{
-			SourceName:  obj,
-			CurrentName: obj,
-		}
-		parent := ""
-		if source.DBType == POSTGRESQL || source.DBType == YUGABYTEDB {
-			parent = source.DB().ParentTableOfPartition(tuple)
-		}
-		if parent == "" {
-			tuple, err = namereg.NameReg.LookupTableName(fmt.Sprintf("%s.%s", schema, table))
-			if err != nil {
-				utils.ErrExit("lookup for table name failed err: %s: %v", table, err)
-			}
+		obj = sqlname.NewObjectNameWithQualifiedName(source.DBType, defaultSchemaName, qualifiedObjectStr)
+	}
+	//get the name tupe case its leaf partition return a handcrafted NameTuple else return  from nameReg lookup
+	var err error
+	tuple := sqlname.NameTuple{
+		SourceName:  obj,
+		CurrentName: obj,
+	}
+	parent := source.DB().ParentTableOfPartition(tuple)
+
+	if parent == "" {
+		tuple, err = namereg.NameReg.LookupTableName(fmt.Sprintf("%s.%s", obj.SchemaName, obj.Unqualified.Unquoted))
+		if err != nil {
+			return sqlname.NameTuple{}, fmt.Errorf("lookup for table name failed err: %s: %v", obj.Unqualified, err)
 		}
-		fullTableList = append(fullTableList, tuple)
 	}
-	excludeTableList := extractTableListFromString(fullTableList, source.ExcludeTableList, "exclude")
-	if len(excludeTableList) > 0 {
-		//TODO: avoid duplicate call to this function and optimize this function later
-		_, excludeTableList, err = addLeafPartitionsInTableList(excludeTableList, true)
+	return tuple, nil
+}
+
+func applyTableListFlagsOnFullListAndAddLeafPartitions(fullTableList []sqlname.NameTuple, tableListViaFlag string, excludeTableListViaFlag string) ([]sqlname.NameTuple, error) {
+	var err error
+	var includeTableList, excludeTableList []sqlname.NameTuple
+
+	applyFilterAndAddLeafTable := func(flagList string, flagName string) ([]sqlname.NameTuple, error) {
+		flagTableList := extractTableListFromString(fullTableList, flagList, flagName)
+		_, flagTableList, err = addLeafPartitionsInTableList(flagTableList, true)
 		if err != nil {
-			utils.ErrExit("adding leaf partititons to exclude table list: %s", err)
+			return nil, fmt.Errorf("failed to add the leaf partitions in %s table list: %w", flagName, err)
 		}
+		return flagTableList, nil
 	}
-	if source.TableList != "" {
-		tableList = extractTableListFromString(fullTableList, source.TableList, "include")
+
+	if excludeTableListViaFlag != "" {
+		//Apply exclude table list filter if present
+		excludeTableList, err = applyFilterAndAddLeafTable(excludeTableListViaFlag, "exclude")
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	includeTableList = fullTableList
+	if tableListViaFlag != "" {
+		//Apply include table list filter if present
+		includeTableList, err = applyFilterAndAddLeafTable(tableListViaFlag, "include")
+		if err != nil {
+			return nil, err
+		}
 	} else {
-		tableList = fullTableList
+		//this is only for filtering the non-leaf and non-root tables - the mid level partitioned table from fullTableList
```

Comment:
To remove non-leaf and non-root, you're still calling `addLeafPartitionsInTableList`? That sounds a bit weird. Does that function ignore all leafs, and then re-add them?

Reply:
`addLeafPartitionsInTableList ` has a boolean parameter `addAllLeafPartitions` in case this is false, that function will consider that table-list to be and exhaustive one already with all the tables (root, middle partition, leaf partitions, so on..) and it will just do the filtering to remove the middle level partitions and just the keep the leaf and root tables in the list.

Reply:
nit: write a separate specific function to remove the non-leaf/non-root tables. 
Add a TODO if defering.